include/circt/Dialect/Calyx/CalyxControl.td

```diff
@@ -372,3 +376,45 @@ def StaticRepeatOp : CalyxContainer<"static_repeat", [
   ];
 
 }
+
+def InvokeOp : CalyxOp<"invoke", [
+                        ControlLike,
+                        SameVariadicOperandSize
+                        ]> {
+  let summary = "Calyx Invoke";
+  let description = [{
+    calyx.invoke is similar to the behavior of a function 
+    call, which invokes a given component.
+
+    The 'callee' attribute is the name of the component,
+    the 'ports' attribute The ports attribute specifies 
+    the input port of the component when it is invoked, 
+    and the 'inputs' attribute specifies the assignment on 
+    the corresponding port.
+
+    ```mlir
+      %id.in, ... = calyx.instance @id of @identity : i32, ...
+      ...
+      calyx.control {
+        calyx.seq {
+          calyx.invoke @id(%id.in = %c1_10) -> (i32)
```

Comment:
Let's include both in and out connections in the example, since this has a different syntax than the native Calyx language.